dev/CsvBulkLoader.php

```diff
@@ -47,8 +56,99 @@ public function preview($filepath) {
 	/**
 	 * @param string $filepath
 	 * @param boolean $preview
+	 *
+	 * @return null|BulkLoader_Result
 	 */
 	protected function processAll($filepath, $preview = false) {
+		$files = $this->splitFile($filepath);
+
+		$result = null;
+
+		foreach ($files as $file) {
+			$next = $this->processChunk($file);
+
+			if ($result instanceof BulkLoader_Result) {
+				$result->merge($next);
+			} else {
+				$result = $next;
+			}
+
+			@unlink($file);
+		}
+
+		return $result;
+	}
+
+	/**
+	 * Splits a large file up into many smaller files.
+	 *
+	 * @param string $path Path to large file to split
+	 * @param int $lines Number of lines per file
+	 *
+	 * @return array List of file paths
+	 */
+	protected function splitFile($path, $lines = null) {
+		ini_set('auto_detect_line_endings', true);
+
+		if (!is_int($lines)) {
+			$lines = $this->config()->get("lines");
+		}
+
+		$new = $this->getNewSplitFileName();
+
+		$to = fopen($new, 'w+');
+		$from = fopen($path, 'r');
+
+		// pop the headers and write them to the new file
+		$header = fgets($from, 4096);
```

Comment:
CSVBulkloader doesn't mandate header rows (see `CSVBulkLoader-> hasHeaderRow`), so this could lead to duplicate data imports.